chaoscenter/web/src/views/ChaosInfrastructureReferenceField/ChaosInfrastructureReferenceField.tsx

```diff
@@ -153,79 +203,163 @@ function ChaosInfrastructureReferenceFieldView({
+          <Layout.Vertical height={'100%'}>
+            <Layout.Horizontal flex width={'100%'} padding={{ bottom: 'small' }} border={{ bottom: true }}>
+              <Container
+                width={'50%'}
+                padding={{ left: 'medium', right: 'medium' }}
+                flex={{ justifyContent: 'space-between' }}
+              >
+                <Text color={Color.GREY_800} font={{ variation: FontVariation.H5 }}>
+                  {getString('environments')}
+                </Text>
+                <Text color={Color.GREY_800} font={{ variation: FontVariation.H5 }}>
+                  {getString('infrastructures')}
+                </Text>
+              </Container>
+              <ExpandingSearchInput
+                alwaysExpanded
+                throttle={500}
+                placeholder={getString('search')}
+                onChange={e => setSearchInfrastructure(e)}
+              />
+            </Layout.Horizontal>
+            <Layout.Horizontal height={'calc(87% - 10px)'}>
+              <Layout.Horizontal
+                width={'30%'}
+                padding={{ bottom: 'medium', top: 'small' }}
+                className={css.gap4}
+                border={{ right: true }}
+              >
+                <Layout.Vertical
+                  width={'30%'}
+                  padding={{ left: 'medium', right: 'small', top: 'small', bottom: 'medium' }}
+                  className={css.agentList}
+                >
+                  <Container
+                    flex
+                    padding="small"
+                    className={cx(css.listEnvContainer, activeEnv === 'all' && css.activeEnv)}
+                    onClick={() => {
+                      setEnvID('all');
+                      setActiveEnv('all');
+                    }}
+                  >
+                    <div className={css.itemEnv}>
+                      <Layout.Horizontal padding={{ left: 'small' }} spacing="medium" className={css.leftInfo}>
+                        <Text lineClamp={1} color={Color.GREY_800} font={{ variation: FontVariation.H6 }}>
+                          {getString('all')}
+                        </Text>
+                      </Layout.Horizontal>
+                      <Text
+                        font={{ variation: FontVariation.SMALL }}
+                        color={activeEnv === 'all' ? Color.WHITE : Color.PRIMARY_7}
+                        background={activeEnv === 'all' ? Color.PRIMARY_7 : Color.PRIMARY_BG}
+                        height={24}
+                        width={24}
+                        flex={{ alignItems: 'center', justifyContent: 'center' }}
+                        className={css.rounded}
+                      >
+                        {allInfrastructureLength}
+                      </Text>
+                    </div>
+                  </Container>
+                  {environmentList.map(env => listingEnvironment({ env: env }))}
```

Comment:
mostly we want to use react components with keys rather that functions that return a jsx element